### PR TITLE
Add check_mode tests and test docs

### PR DESCRIPTION
**Pull request by mdellweg**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,16 +1,32 @@
...
+ * The third part, again for `localhost`, provides the the opportunity to undo part one.
+
+In the first and third parts, no recording takes place.
+During playback, only the part in the middle is executed.
+
+To run the tests, you can either call `make test`, or `make test_<playbook_name>` to only run a specific one.
+
+To (re-)record tests, you first need to setup a pulp instance (pulplift is recommended here).
+With it's connection details configured in `tests/playbooks/vars/server.yaml`, you can run `make record_<playbook_name>`.
+For this step, using python virtual environments is recomended.
```
Comment (mdellweg) [inline README.md]: cc @Timoses

File: README.md [hunk 2]
```diff
@@ -1,16 +1,32 @@
...
+It has been wildly inspired by the sister project [foreman-ansible-modules](https://github.com/theforeman/foreman-ansible-modules).
 
 Testing
 ---
 
-Testing infrastructure has been wildly inspired by the sister project [foreman-ansible-modules](https://github.com/theforeman/foreman-ansible-modules).
-Look at the documentation there for now.
+Testing is done by running handcrafted playbooks while playing back prerecorded server answers.
+
+Those playbooks are meant to test one specific module, and are usually structured in three sections:
```
Comment (Timoses) [inline README.md]: "The playbooks under `tests/playbooks`"?

File: README.md (hunk 1 above)
Comment (Timoses) [reply README.md]: recom**m**ended

(missing an `m`).

**Comment by Timoses** [comment]
Looks good ! 👍

File: README.md (hunk 2 above)
Comment (mdellweg) [reply README.md]: good catch!

**Pull request merged**